packages/quantic/force-app/main/default/lwc/quanticResult/quanticResult.js

```diff
@@ -80,7 +106,11 @@ export default class QuanticResult extends LightningElement {
   };
 
   render() {
-    const template = this.resultTemplatesManager.selectTemplate(this.result);
+    const result = {
+      ...this.result,
+      raw: {...this.result.raw, quantic__resultTemplateId: this.templateId},
+    };
+    const template = this?.resultTemplatesManager?.selectTemplate(result);
```

Comment:
I think that it's a more complex solution to implement  since it includes changes in Headless while adding, imo, little to no value comparing to the actual implementation in this PR where we are using the existing Headless template manager and achieving the same goal without having to add changes in Headless.

Reply:
I just don't like adding data to the result object just for the purpose of templating.
IMO templating data should be in a template object, and result fields should be in the result object :D 

But I don't have a clear view on how complex it would create a modification in headless just for that. Maybe it's just simpler that way.

To be honest I don't see a situation where this would hurt us to add a field to the results.

Reply:
As discussed with @erocheleau  we will create a jira to add the method `selectTemplateById` in the Headless result template manager.
For now we will continue using the current implementation of this PR.